training/src/main/scala/com/airbnb/aerosolve/training/DecisionTreeTrainer.scala

```diff
@@ -72,6 +72,15 @@ object DecisionTreeTrainer {
         .filter(x => x.contains(rankKey))
         .take(candidateSize)
 
+    val ex = examples(0)
+    val numFeatures = LinearRankerUtils.getNumFeatures(ex, rankKey)
+    val maxFeatures : Int = config.getString(key + ".max_features") match {
```

Comment:
It's better to use `Try().getOrElse` to get the training parameter, otherwise this will break some of our pricing jobs that are using decision tree model